src/client/apiclient.ts

```diff
@@ -21,14 +22,29 @@ export const getCompanies =
     };
 
 export const getDissolvedCompanies =
-async (apiKey: string, companyName: string, requestId, searchType: string, page: number, searchBefore: string | null, searchAfter: string | null, size: number | null): Promise<DissolvedCompaniesResource> => {
-    const api = createApiClient(apiKey, undefined, API_URL);
-    const startIndexOffset = (page * DISSOLVED_SEARCH_NUMBER_OF_RESULTS) - DISSOLVED_SEARCH_NUMBER_OF_RESULTS;
-    const companiesResource: Resource<DissolvedCompaniesResource> =
-        await api.dissolvedSearch.getCompanies(companyName, requestId, searchType, startIndexOffset, searchBefore, searchAfter, size);
-    if (companiesResource.httpStatusCode !== 200 && companiesResource.httpStatusCode !== 201) {
-        throw createError(companiesResource.httpStatusCode, companiesResource.httpStatusCode.toString());
-    }
-    logger.info(`Get company name dissolved search results, company_name=${companyName}, status_code=${companiesResource.httpStatusCode}`);
-    return companiesResource.resource as DissolvedCompaniesResource;
-};
+    async (apiKey: string, companyName: string, requestId, searchType: string, page: number, searchBefore: string | null, searchAfter: string | null, size: number | null): Promise<DissolvedCompaniesResource> => {
+        const api = createApiClient(apiKey, undefined, API_URL);
+        const startIndexOffset = (page * DISSOLVED_SEARCH_NUMBER_OF_RESULTS) - DISSOLVED_SEARCH_NUMBER_OF_RESULTS;
+        const companiesResource: Resource<DissolvedCompaniesResource> =
+            await api.dissolvedSearch.getCompanies(companyName, requestId, searchType, startIndexOffset, searchBefore, searchAfter, size);
+        if (companiesResource.httpStatusCode !== 200 && companiesResource.httpStatusCode !== 201) {
+            throw createError(companiesResource.httpStatusCode, companiesResource.httpStatusCode.toString());
+        }
+        logger.info(`Get company name dissolved search results, company_name=${companyName}, status_code=${companiesResource.httpStatusCode}`);
+        return companiesResource.resource as DissolvedCompaniesResource;
+    };
+
+export const getAdvancedCompanies =
+    async (apiKey: string, companyNameIncludes: string | null, companyNameExcludes: string | null, location: string | null, incorporatedFrom: string | null,
```

Comment:
A potential future improvement, we could map these values to an object in the controller layer similar to how we have done in the API and then extract the values in the SDK. Would mean reducing the number of places needed to change when adding a new request param. Something for another day however.

Reply:
cheers, yeah that's a good idea